client/my-sites/plan-features/header.jsx

```diff
@@ -14,12 +14,44 @@ import {
 	PLAN_PREMIUM,
 	PLAN_BUSINESS
 } from 'lib/plans/constants';
+import currencyFormatter, { findCurrency } from 'lib/currency-formatter';
+
+function currencyFormatOptions( defaults, showDecimal = true ) {
+	const options = { code: defaults.code };
+	const space = defaults.spaceBetweenAmountAndSymbol ? '\u00a0' : '';
+
+	options.format = `<sup class="plan-features__header-currency-symbol">%s</sup>${ space }%v`;
+
+	if ( ! defaults.symbolOnLeft && showDecimal ) {
+		options.format = `%v</sup>${ space }<sup class="plan-features__header-currency-symbol">%s</sup>`;
+	} else if ( ! defaults.symbolOnLeft ) {
+		options.format = `%v${ space }<sup class="plan-features__header-currency-symbol">%s</sup>`;
+	} else if ( defaults.symbolOnLeft && showDecimal ) {
+		options.format = `<sup class="plan-features__header-currency-symbol">%s</sup>${ space }%v</sup>`;
+	}
+
+	if ( showDecimal ) {
+		options.decimal = `<sup class="plan-features__header-cents">${ defaults.decimalSeparator }`; //open sup is closed by format string
+	} else {
+		options.precision = 0;
+	}
+	return options;
+}
 
 class PlanFeaturesHeader extends Component {
 
+	getPrice( rawPrice, code ) {
+		const defaults = findCurrency( code );
+		const priceHTML = currencyFormatter( rawPrice, currencyFormatOptions( defaults, rawPrice !== 0 ) );
+		/*eslint-disable react/no-danger*/
+		return (
+			<h4 className="plan-features__header-price" dangerouslySetInnerHTML={ { __html: priceHTML } } />
```

Comment:
😱  Reiterating @dmsnell's comment, can we really not avoid the `dangerouslySetInnerHTML` ?